### ISSUE TITLE
Number of realizations value to be available for a forward model step

### ISSUE THREAD

**Issue by joargr** [equinor/ert#8852]
**Is your feature request related to a problem? Please describe.**
For the RMS forward model step, we are calling the `validate_pre_experiment` hook. This validates some configuration and files used by RMS. Having access to the ERT configuration `NUM_REALIZATIONS` value in the validation means that we can perform an additional step in the validation. If this step fails we can raise an exception and abort the experiment, before any realizations are started.

**Describe the solution you'd like**
The ERT configuration `NUM_REALIZATIONS` value could be made available in `fm_step_json["environment"]`, where values like `_ERT_REALIZATION_NUMBER` are available today.

**Describe alternatives you've considered**
A similar validation step can be done today by using `fm_step_json["environment"]["_ERT_REALIZATION_NUMBER"]`, but this validation would then be done prior to each realization, possibly leading to some realizations having successful validations but other realizations failing the validation. It is preferred that if there are some realizations with failed validations then no realizations should be run, and the experiment should be aborted. This is not possible today.

**Comment by eivindjahren** [comment]
Since users can decide which realizations to run in the GUI before pressing the run button, we would have to add a separate validation method that validates when the run button is pressed. Would this work for you?

**Comment by KetilN** [comment]
Yes, I think in general that we need a separate validation method when the run button is pressed since the extra information at this point also can be relevant. This will then be at a later validation point (in time sequence) then when validating the forward models today, right ? Can this separate validation method also call the "validation method on run button click" per forward model like the "first/primary validation method" ?

**Comment by joargr** [comment]
> Since users can decide which realizations to run in the GUI before pressing the run button, we would have to add a separate validation method that validates when the run button is pressed. Would this work for you?

Currently the two validation hooks `validate_pre_experiment` and `validate_pre_realization_run` are available for an ERT plugin. For our use case of validating the random seeds file `validate_pre_experiment` seems to be the best to use, as it should execute if there multiple realizations to be run. If the user has selected one or more runs in the GUI and is then pressing the run button, will `validate_pre_experiment` also be executed? In other words, will `validate_pre_experiment` always be executed as long as there are multiple realizations to be run? And what if only one realization has been selected, will `validate_pre_experiment` still be executed?

For this use case I don't think we need to know if realizations are run due to the user having pressed the run button in the GUI, although other use cases might have that need. The important thing about the use case is to verify that the random seeds has a valid format, with a number of seed values exceeding any number of realizations that could be run as part of an experiment. Even if the user only wants to run one realization, we want to a failed validation to abort an experiment and to not have any realizations start.
